src/lib.rs

```diff
@@ -650,21 +668,35 @@ impl Signer for ExternalAccountKey {
     }
 }
 
-fn nonce_from_response(rsp: &Response<Body>) -> Option<String> {
+fn nonce_from_response(rsp: &Response<Incoming>) -> Option<String> {
     rsp.headers()
         .get(REPLAY_NONCE)
         .and_then(|hv| String::from_utf8(hv.as_ref().to_vec()).ok())
 }
 
 #[cfg(feature = "hyper-rustls")]
-struct DefaultClient(hyper::Client<hyper_rustls::HttpsConnector<HttpConnector>>);
+struct DefaultClient(
+    hyper_util::client::legacy::Client<
+        hyper_rustls::HttpsConnector<HttpConnector>,
+        BoxBody<Bytes, Error>,
```

Comment:
Why do we need to use `BoxBody` here? What are we abstracting over?

Reply:
Perhaps we could just use `Full` and provide it with no bytes as a substitute for `Empty`, I'm not sure if that breaks any invariants that `Full` requires, it does seem to go against the grain a little from how the types are expected to be used but I guess it does allow us to avoid trait boxing. Another alternative could be to make our own body enum with `Full` and `Empty` variants using enum dispatch technique.

Reply:
Using an empty `Full` seems like a good solution here.

Reply:
Their `Body` implementations are slightly different, Even when there is no data, [`Full` returns a `Some` with a `Frame` ](https://docs.rs/http-body-util/latest/src/http_body_util/full.rs.html#39-44), and [`Empty` returns a `None`](https://docs.rs/http-body-util/latest/src/http_body_util/empty.rs.html#28-33), I guess it's up to the `HttpClient` implementation whether this would make any difference.

Reply:
I've swapped to using `Full` anyway, and conveniently it resolved the issue https://github.com/instant-labs/instant-acme/pull/50#discussion_r1650033771